code/modules/mining/lavaland/loot/ashdragon_loot.dm

```diff
@@ -177,10 +177,15 @@
 	var/create_delay = 30
 	var/reset_cooldown = 50
 	var/timer = 0
+	var/timerMalfunction = 0
 	var/banned_turfs
+	var/datum/effect_system/spark_spread/spark_system
 
 /obj/item/lava_staff/New()
 	. = ..()
+	spark_system = new /datum/effect_system/spark_spread
+	spark_system.set_up(5, 0, src)
+	spark_system.attach(src)
```

Comment:
`do_sparks` essentially covers what you've trying to accomplish here (minus doing `.attach(src)` because it doesn't need to).
```js
/proc/do_sparks(n, c, source)
	var/datum/effect_system/spark_spread/sparks = new
	sparks.set_up(n, c, source)
	sparks.autocleanup = TRUE
	sparks.start()
```
So I recommend either: removing these lines, and the `var/datum/effect_system/spark_spread/spark_system` reference you've created on line 182. 

Or, instead of calling `do_sparks(5,FALSE,user)` on line 204, call `spark_system.start()`

Reply:
Gonna go with do_sparks(5, FALSE, user).